pservice/lib/libpdo_enclave/contract_enclave_mrenclave.cpp.template

```diff
@@ -0,0 +1,18 @@
+/* Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "types.h"
+
+HexEncodedString ALLOWED_CONTRACT_ENCLAVE_MR_ENCLAVE = "MR_ENCLAVE_PLACEMARK";
```

Comment:
Did some poking around and I'm pretty happy with this. If you don't build the eservice, the pservice build succeeds in SIM mode but not HW (because contract_enclave_mrenclave.cpp doesn't exist). The error you get is : 

```
secret_enclave.cpp:(.text+0x1f41): undefined reference to `ALLOWED_CONTRACT_ENCLAVE_MR_ENCLAVE'
collect2: error: ld returned 1 exit status
```

I think this is good enough. I wish it was easy to make this spit out an error like "compile the eservice first!" but I don't know of an easy way to do that.

Reply:
That's a nice suggestion actually!
How about putting something like the following in the template:
#define IS_MR_ENCLAVE_SET 0
#if SGX_SIMULATOR && !IS_MR_ENCLAVE_SET
#error "compile the eservice first!"
#endif
I would expect that the IS_MR_ENCLAVE_SET is changed accordingly.

Reply:
Good idea - and easy to add. This will make the error a lot more clear - right now it does actually fail but the above error isn't clear.

Reply:
Fixed. CMake checks if the file exists in HW mode.

Reply:
Added check to cmake file. Not checks if file exists when in HW mode.